Sources/Lifecycle/Lifecycle.swift

```diff
@@ -201,6 +201,11 @@ public struct ServiceLifecycle {
         self.underlying = ComponentLifecycle(label: self.configuration.label, logger: self.configuration.logger)
         // setup backtraces as soon as possible, so if we crash during setup we get a backtrace
         self.installBacktrace()
+        if self.configuration.shutdownSignal != nil {
+            self.register(label: "Shutdown hooks",
+                          start: .sync(self.setupShutdownHook),
+                          shutdown: .none)
+        }
```

Comment:
should this rather be `self.installShutdownSignalHooks`? 

It'd be the same style as install backtrace and avoid having to much logic in the init function itself

Reply:
this is great @hassila!

+1 to abstract this to an  function "installShutdownSignalHook" function

Reply:
I did a slight reiteration along those lines while also adding support for proper cancellation fo all dispatch signal sources during shutdown, please have a look again.